### PR TITLE
v2.x Python dep updates

### PR DESCRIPTION
**Pull request by ddymko**
## Description
Cherry-picked the following two commits to update some of the python Libs that have been updated in v3.

- https://github.com/emissary-ingress/emissary/pull/4385/commits/3942b8a2bef69f38dd01470590dfc12c560bfe79
- https://github.com/emissary-ingress/emissary/pull/4385/commits/c6889c50a9b6d5df6d8770aea96ea40de802337e

## Related Issues
List related issues.

## Testing
CI goes green

## Checklist

 - [ ] I made sure to update `CHANGELOG.md`.

   Remember, the CHANGELOG needs to mention:
    + Any new features
    + Any changes to our included version of Envoy
    + Any non-backward-compatible changes
    + Any deprecations

 - [x] This is unlikely to impact how Ambassador performs at scale.

   Remember, things that might have an impact at scale include:
    + Any significant changes in memory use that might require adjusting the memory limits
    + Any significant changes in CPU use that might require adjusting the CPU limits
    + Anything that might change how many replicas users should use
    + Changes that impact data-plane latency/scalability

 - [x] My change is adequately tested.

   Remember when considering testing:
    + Your change needs to be specifically covered by tests.
       + Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
    + You also need to make sure that the _entire area being changed_ has adequate test coverage.
       + If existing tests don't actually cover the entire area being changed, add tests.
       + This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
    + We should lean on the bulk of code being covered by unit tests, but...
    + ... an end-to-end test should cover the integration points

 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.

 - [ ] The changes in this PR have been reviewed for security concerns and adherence to security best practices.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -99,3 +99,52 @@ deploy-only: preflight-dev-kubeconfig $(tools/kubectl) build-output/yaml-$(patsu
 	@printf "$(GRN)Your ambassador image:$(END) $(BLD)$$($(tools/kubectl) --kubeconfig $(DEV_KUBECONFIG) get -n ambassador deploy ambassador -o 'go-template={{(index .spec.template.spec.containers 0).image}}')$(END)\n"
 	@printf "$(GRN)Your built image:$(END) $(BLD)$$(sed -n 2p docker/$(LCNAME).docker.push.remote)$(END)\n"
 .PHONY: deploy-only
+
+##############################################
```
Comment (LanceEa) [inline Makefile:103]: Where is all this telepresence stuff getting pulled in from?

File: Makefile [hunk 2]
```diff
@@ -99,3 +99,52 @@ deploy-only: preflight-dev-kubeconfig $(tools/kubectl) build-output/yaml-$(patsu
...
+$(RUN_EMISSARY_AGENT):
+	@test -e $(EMISSARY_AGENT_ENV) || echo "Environment file $(EMISSARY_AGENT_ENV) does not exist, please run 'make intercept-emissary-agent' to create it."
+	echo 'AES_LOG_LEVEL=debug AES_SNAPSHOT_URL=http://ambassador-admin.ambassador:8005/snapshot-external AES_DIAGNOSTICS_URL="http://ambassador-admin.ambassador:8877/ambassador/v0/diag/?json=true" AES_REPORT_DIAGNOSTICS_TO_CLOUD=true go run ./cmd/busyambassador agent' >> $(RUN_EMISSARY_AGENT)
+	chmod a+x $(RUN_EMISSARY_AGENT)
+
+.PHONY: irun-emissary-agent
+irun-emissary-agent: bin/run-emissary-agent.sh ## Run emissary-agent using the environment variables fetched by the intercept.
+	bin/run-emissary-agent.sh
+
+## Helper target for setting up local dev environment when working with python components
```
Comment (LanceEa) [inline Makefile:135]: This part was pulled in from `fb29f474061115c5c506038be80c70e3c62ada18` but the rest of it is showing up as changed too?

File: Makefile (hunk 1 above)
Comment (ddymko) [reply Makefile:103]: I left it in due to the fact it would match up with what is currently on master

Comment (LanceEa) [reply Makefile:103]: Ok, yeah I reviewed them and they don't hurt and will make back porting other changes in the future easier.

**Review by LanceEa — APPROVED** [review]

**Review by AliceProxy — APPROVED** [review]

**Pull request merged**